MinecraftClient/McTcpClient.cs

```diff
@@ -1558,6 +1566,61 @@ public void OnUpdateHealth(float health, int food)
                     ConsoleIO.WriteLogLine("You are dead. Type /respawn to respawn.");
                 }
             }
+            if (Settings.AutoEat)
+            {
+                if (food <= HungerThreshold || (food < 20 && health < 20))
+                {
+                    Eating = true;
+                    FindFoodAndEat();
+                }
+                // keep eating until full
+                if (food < 20 && Eating)
+                {
+                    FindFoodAndEat();
+                }
+                if (food >= 20 && Eating)
+                {
+                    Eating = false;
+                    ChangeSlot(LastSlot);
+                }
+            }
+        }
+
+        public void OnHeldItemChange(byte slot)
+        {
+            CurrentSlot = slot;
+        }
+
+        /// <summary>
+        /// Try to find food in the hotbar and eat it
+        /// </summary>
+        /// <returns>True if found</returns>
+        public bool FindFoodAndEat()
```

Comment:
Since inventories are reachable from ChatBots, I  _think_ this also could be moved to the AutoEat bot 🤔